packages/react-dom-bindings/src/server/fizz-instruction-set/ReactDOMFizzInstructionSet.js

```diff
@@ -42,16 +42,121 @@ export function clientRenderBoundary(
   }
 }
 
-export function completeBoundaryWithStyles(
+// The following two functions should be almost identical. The only differences
+//  are the values used for `completeBoundaryImpl` and `resourceMap`.
+// completeBoundaryWithStylesInlineLocals is used by the Fizz external runtime, which
+//  bundles together all Fizz instruction functions (and is able to reference / rename
+//  completeBoundary and resourceMap as locals).
+// completeBoundaryWithStylesInlineGlobals is used by the Fizz inline script writer,
+//  which sends Fizz instruction functions on an as-needed basis. This version needs
+//  to reference completeBoundary($RC) and resourceMap($RM) as globals.
+//
+// Ideally, Closure would take care of inlining a shared implementation, but I couldn't
+//  figure out a zero-overhead inline due to lack of a @inline compiler directive.
+export function completeBoundaryWithStylesInlineLocals(
+  resourceMap,
+  suspenseBoundaryID,
+  contentID,
+  styles,
+) {
+  const completeBoundaryImpl = completeBoundary;
+  const precedences = new Map();
+  const thisDocument = document;
+  let lastResource, node;
+
+  // Seed the precedence list with existing resources
```

Comment:
Did you try extracting smaller parts of the function out into a shared function? My guess is if you extract everything between lines 67-140 into a separate function, Closure would be able to inline it.

Reply:
> Did you try extracting smaller parts of the function out into a shared function? My guess is if you extract everything between lines 67-140 into a separate function, Closure would be able to inline it.

Yep I tried extracting just those lines (with no luck). I played around with compiling some much smaller functions, and it looks like Closure does inline small ones (so it's not that Closure decides to not inline accesses to globals or something). 
```
function completeBoundaryWithStylesShared(
  resourceMap,
  dependencies,
  styles,
) {
  // ...
}

export function completeBoundaryWithStylesInlineGlobals(
  suspenseBoundaryID,
  contentID,
  styles,
) {
  const completeBoundaryImpl = window['$RC'];
  const resourceMap = window['$RM'];
  const dependencies = [];

  completeBoundaryWithStylesShared(resourceMap, dependencies, styles);

  Promise.all(dependencies).then(
    completeBoundaryImpl.bind(null, suspenseBoundaryID, contentID, ''),
    completeBoundaryImpl.bind(
      null,
      suspenseBoundaryID,
      contentID,
      'Resource failed to load',
    ),
  );
}
```


I'm not sure how maintainable it is to try to find "at which point" Closure stops inlining functions. The code duplication does feel silly -- an alternative would be to use `rollup`'s `replace`, or even import from a forked module. However these feel quite heavy handed (and fragile, since this would need to be added in `build.js`, `generate-inline-fizz-runtime.js`, and `fizzTestUtils.js`.)
e.g.
```
let completeBoundaryImpl;
if (COMPILING_INLINE_JS) {
  completeBoundaryImpl = window['$RC'];
} else {
  completeBoundaryImpl = completeBoundary;
}
```

Reply:
It feels extra silly, since Closure will actually inline the variables correctly!

```
function completeBoundaryWithStylesShared(
  completeBoundaryImpl,
  resourceMaps,
  suspenseBoundaryID,
  contentID,
  styles,
) {
  ...
}
window['$RR'] = function (
  suspenseBoundaryID,
  contentID,
  styles,
) {
  completeBoundaryWithStylesShared(
    window['$RC'],
    window['$RM'],
    suspenseBoundaryID,
    contentID,
    styles,
  );
}

```
gets compiled to this. Notice that the first two parameters (`completeBoundaryImpl` and `resourceMaps`) get inlined, but this useless wrapper function is still kept around.
```
function w(g,h,n){...} $RR=function(g,h,n){w(g,h,n)};

Reply:
Hmm my guess is that its not size alone, because there are some really large functions in the reconciler that Closure manages to inline.

How about if you move the non-shared function wrappers out of ReactDOMFizzInstructionSet.js and into the runtime module that imports it? Then, from Closure's perspective, it will only see a single call of the shared function. Because the other call is in the other runtime.

Reply:
I've tried this as well (and more drastically, only declaring and exporting the version referencing globals + the shared function), with no luck. 😟

Reply:
That's surprising, huh